src/kaleidoscope/plugin/OneShot.h

```diff
@@ -68,45 +62,85 @@ class OneShot : public kaleidoscope::Plugin {
   static void disableStickabilityForModifiers();
   static void disableStickabilityForLayers();
 
+  // --------------------------------------------------------------------------
+  // Global test functions
+
+  static bool isActive();
+  static bool isSticky();
+  static bool isPressed();
+
+  // --------------------------------------------------------------------------
+  // Single-key test functions
+
+  static bool isOneShotKey(Key key) {
+    return (key.getRaw() >= kaleidoscope::ranges::OS_FIRST &&
+            key.getRaw() <= kaleidoscope::ranges::OS_LAST);
+  }
+  static bool isActive(Key key);
+  static bool isSticky(Key key);
+  static bool isPressed(Key key);
   static bool isStickable(Key key);
 
+  // --------------------------------------------------------------------------
+  // Utility function for other plugins to cancel OneShot keys
+  static void cancel(bool with_stickies = false);
+
+  // --------------------------------------------------------------------------
+  // Vestigial functions?
+  void inject(Key mapped_key, uint8_t key_state);
   static bool isModifierActive(Key key);
 
+  // --------------------------------------------------------------------------
+  // Configuration variables (should probably be private)
+  static uint16_t time_out;
+  static uint16_t hold_time_out;
+  static int16_t double_tap_time_out;
+
+  // --------------------------------------------------------------------------
+  // Plugin hook functions
+
+  EventHandlerResult onKeyswitchEvent(Key &mapped_key, KeyAddr key_addr, uint8_t key_state);
   EventHandlerResult beforeReportingState();
   EventHandlerResult afterEachCycle();
-  EventHandlerResult onKeyswitchEvent(Key &mapped_key, KeyAddr key_addr, uint8_t keyState);
-
-  void inject(Key mapped_key, uint8_t key_state);
 
  private:
+
+  // --------------------------------------------------------------------------
+  // Constants
   static constexpr uint8_t ONESHOT_KEY_COUNT = 16;
-  typedef struct {
+  static constexpr uint8_t ONESHOT_MOD_COUNT = 8;
```

Comment:
Instead of doing math (e.g. `ONESHOT_KEY_COUNT / 2`), I think it's clearer and more robust to use two constants. Iteration indices can start and end at whichever constant is appropriate, and if we wanted to change the number of supported OneShot layers, it wouldn't require changing anything but `ONESHOT_KEY_COUNT`.